emr_launcher/util.py

```diff
@@ -230,3 +232,12 @@ def adg_trim_steps_for_incremental(cluster_config, snapshot_type):
             if steps[step_count][NAME_KEY] == SNS_NOTIFICATION_STEP:
                 del cluster_config[STEPS][step_count]
                 break
+
+
+def adg_trim_steps_for_full(cluster_config, snapshot_type):
```

Comment:
This will only ever remove one step with "build-day-1-" in it. Is that right?

If not, you will need to remove the break and operate the loop on a deep copy of `cluster_config[STEPS]`. Then set `cluster_config[STEPS]` back to the changed step array.

Can you ensure there is a unit test where multiple  "build-day-1-" steps are removed if you do need this functionality please.

Reply:
fixed it